### PR TITLE
Branch to update lcd functions and state name functions

### PR DESCRIPTION
**Pull request by TaylorSmith28**
Update states to have lcd_clear()

### CODE REVIEW

**Comment by TaylorSmith28** [comment]
Refactored UI state tests:
 - Removed all instances of reset mock
 - Renamed all variables and functions to follow the snake_case naming style
 - Added test function docstrings to each test
 - Added document docstrings to each test file
 - Separated test functions into multiple tests if deemed necessary

File: .github/actions/spelling/advice.md [hunk 1]
```diff
@@ -7,18 +7,18 @@ If items relate to a ...
...
-https://www.regexplanet.com/advanced/perl/) yours before committing to verify it will match your files.
+<https://www.regexplanet.com/advanced/perl/>) yours before committing to verify it will match your files.
 
   `^` refers to the file's path from the root of the repository, so `^README\.md$` would exclude [README.md](
 ../tree/HEAD/README.md) (on whichever branch you're using).
 
 * well-formed pattern.
 
-  If you can write a [pattern](https://github.com/check-spelling/check-spelling/wiki/Configuration-Examples:-patterns) that would match it,
+  If you can write a [pattern](<https://github.com/check-spelling/check-spelling/wiki/Configuration-Examples:-patterns>) that would match it,
```
Comment (prestoncarman) [inline .github/actions/spelling/advice.md:17]: Do not include the `<>` inside the link.

File: .github/actions/spelling/advice.md [hunk 2]
```diff
@@ -7,18 +7,18 @@ If items relate to a ...
...
   try adding it to the `patterns.txt` file.
 
   Patterns are Perl 5 Regular Expressions - you can [test](
-https://www.regexplanet.com/advanced/perl/) yours before committing to verify it will match your lines.
+<https://www.regexplanet.com/advanced/perl/>) yours before committing to verify it will match your lines.
```
Comment (prestoncarman) [inline .github/actions/spelling/advice.md:21]: Do not include the `<>` inside the link.

File: titration/utils/ui_state/update_settings/update_settings.py [hunk 3]
```diff
@@ -1,67 +1,110 @@
...
+        elif self.substate == 2:
+            self.substate += 1
 
-        elif self.subState == 3:
-            if key != "n" and key != "N":
-                self.subState += 1
-            else:
-                self._setNextState(self.previousState, True)
+        elif self.substate == 3:
+            if key == "y" or key == "Y":
```
Comment (prestoncarman) [inline titration/utils/ui_state/update_settings/update_settings.py:60]: Does this switch the menu steps?

File: titration/utils/ui_state/update_settings/update_settings.py [hunk 4]
```diff
@@ -1,67 +1,110 @@
...
-        if self.subState == 1:
-            if key != "n" and key != "N":
-                self.subState += 1
-            else:
-                self.subState += 2
+        Parameters:
+            key (char): the keypad input is used to move through the substates
+        """
+        if self.substate == 1:
+            if key == "Y" or key == "y":
```
Comment (prestoncarman) [inline titration/utils/ui_state/update_settings/update_settings.py:51]: Does this switch the menu steps?

**Review by prestoncarman — CHANGES_REQUESTED** [review]
Please take a look at these comments.

This is a large PR ... I hope we are getting closer having more focused changes that don't involve 40 plus files. Please don't add any more changes to this PR. Only fix issues related to the comments.

File: .github/actions/spelling/advice.md (hunk 1 above)
Comment (TaylorSmith28) [reply .github/actions/spelling/advice.md:17]: The markdown linter has an issue with "bare_urls" and recommended adding angle brackets. I will change it back but the linter may not pass. View this site for more info: https://github.com/updownpress/markdown-lint/blob/master/rules/034-no-bare-urls.md

File: .github/actions/spelling/advice.md (hunk 2 above)
Comment (TaylorSmith28) [reply .github/actions/spelling/advice.md:21]: The markdown linter has an issue with "bare_urls" and recommended adding angle brackets. I will change it back but the linter may not pass. View this site for more info: https://github.com/updownpress/markdown-lint/blob/master/rules/034-no-bare-urls.md

File: titration/utils/ui_state/update_settings/update_settings.py (hunk 3 above)
Comment (TaylorSmith28) [reply titration/utils/ui_state/update_settings/update_settings.py:60]: This does not switch the menu steps. The menu asks for the user to enter "y" before moving to the next state. Prior to my change, a user could have pressed any button to continue. This makes the handle_key function consistent with what the menu is displaying

File: titration/utils/ui_state/update_settings/update_settings.py (hunk 4 above)
Comment (TaylorSmith28) [reply titration/utils/ui_state/update_settings/update_settings.py:51]: This does not switch the menu steps. The menu asks for the user to enter "y" before moving to the next state. Prior to my change, a user could have pressed any button to continue. This makes the handle_key function consistent with what the menu is displaying

**Review by prestoncarman — APPROVED** [review]
LGTM

**Pull request merged**